### ISSUE TITLE
AAAAAAAAAAAAAAAAAAAAA

### ISSUE THREAD

**Issue by ElbryanWyndon** [FunkinCrew/Funkin#977]
im probably dumb but ive been trying forever my dumb ape brain deosnt know how to compile it i did all the things and downloaded all the things and did the api stuff things (using notepad if it makes a difference) i dont know how to code i just wanna play funk game but dumb computer whenevr  i try to compile and do the lime test windows it say "haha heck you `C:/HaxeToolkit/haxe/lib/polymod/git/polymod/format/XMLMerge.hx:189: characters 10-23 : Warning : This typedef is deprecated in favor of haxe.xml.Access
C:/HaxeToolkit/haxe/lib/polymod/git/polymod/format/XMLMerge.hx:189: characters 30-43 : Warning : This typedef is deprecated in favor of haxe.xml.Access
C:\Users\Boohb\OneDrive\Desktop\Funkin-master\export\release\windows\obj>setlocal enabledelayedexpansion
Error: 64bit is not automatically supported for this version of VC. Set HXCPP_MSVC_CUSTOM and manually configure the executable, library and include paths
Error: Error: 64bit is not automatically supported for this version of VC. Set HXCPP_MSVC_CUSTOM and manually configure the executable, library and include paths` and i dont like it and im confuseed and sad i just wanna do mod with dance dance revolution finger edition :(((((

**Comment by Bloopsdoob** [comment]
have you installed Microsoft Visual Studio Community yet?
Instructions are listed in the project's README

**Comment by ElbryanWyndon** [comment]
i just did that i downloaded visual studio 2019 and now it says `C:/HaxeToolkit/haxe/lib/polymod/git/polymod/format/XMLMerge.hx:189: characters 10-23 : Warning : This typedef is deprecated in favor of haxe.xml.Access
C:/HaxeToolkit/haxe/lib/polymod/git/polymod/format/XMLMerge.hx:189: characters 30-43 : Warning : This typedef is deprecated in favor of haxe.xml.Access
C:\Users\Boohb\OneDrive\Desktop\Funkin-master\export\release\windows\obj>setlocal enabledelayedexpansion
Warning: Could not find Visual Studio 2017 VsDevCmd
Missing HXCPP_VARS
Error: Could not automatically setup MSVC`

**Comment by ElbryanWyndon** [comment]
uhhh issue i just checked and it said on the install instructions "Compiling game
Once you have all those installed, it's pretty easy to compile the game. You just need to run 'lime test html5 -debug' in the root of the project to build and run the HTML5 version. (command prompt navigation guide can be found here: https://ninjamuffin99.newgrounds.com/news/post/1090480)

To run it from your desktop (Windows, Mac, Linux) it can be a bit more involved. For Linux, you only need to open a terminal in the project directory and run 'lime test linux -debug' and then run the executible file in export/release/linux/bin. For Windows, you need to install Visual Studio Community 2019. While installing VSC, don't click on any of the options to install workloads. Instead, go to the individual components tab and choose the following:

MSVC v142 - VS 2019 C++ x64/x86 build tools
Windows SDK (10.0.17763.0)
C++ Profiling tools
C++ CMake tools for windows
C++ ATL for v142 build tools (x86 & x64)
C++ MFC for v142 build tools (x86 & x64)
C++/CLI support for v142 build tools (14.21)
C++ Modules for v142 build tools (x64/x86)
Clang Compiler for Windows
Windows 10 SDK (10.0.17134.0)
Windows 10 SDK (10.0.16299.0)
MSVC v141 - VS 2017 C++ x64/x86 build tools
MSVC v140 - VS 2015 C++ build tools (v14.00)
This will install about 22GB of crap, but once that is done you can open up a command line in the project's directory and run lime test windows -debug. Once that command finishes (it takes forever even on a higher end PC), you can run FNF from the .exe file under export\release\windows\bin As for Mac, 'lime test mac -debug' should work, if not the internet surely has a guide on how to compile Haxe stuff for Mac." but uh my entire hard drive is 54 gigabytes so is there any way i can fix it and make it not take up all my space i only have 183 mb left

**Comment by actuallyasmartname** [comment]
183 MB? God, that's seriously a small amount of free space. You're gonna need a new hard drive.

**Comment by NotHyper-474** [comment]
Can't you just play it (or download from) itch.io or Newgrounds? It would be 100x more easy to play from there than having to compile the source code.

**Comment by actuallyasmartname** [comment]
He wants to do a mod so he can't just get it off of newgrounds or itch.io

**Comment by VMGuy23** [comment]
Can't you just download from itch.io and make the mod using the compiled game?

**Comment by Bloopsdoob** [comment]
most mods come with their own exe, why you may ask?
the game cant easily be modded out of the box, using the source code is one of the best ways to get into the game's code, which makes modding easy with a lil coding experience.
compiling the game yourself is a must if you wanna go beyond the scope of merely replacing assets

**Comment by actuallyasmartname** [comment]
> Can't you just download from itch.io and make the mod using the compiled game?

True.

**Comment by ElbryanWyndon** [comment]
Also I'm using a pre-built all in one windows tablet. I can swap out a hard drive, is there any alternatives so i dont have to delete like everything on my hard drive

**Comment by ElbryanWyndon** [comment]
still cant download it what do i do i dont wanna spend money for a new pc or an external hard drive im 12 years old i just wanna put one of my crappy OC's into funk game :(

**Comment by Steviegt6** [comment]
If you're out of options, then you're out of options.

**Comment by ElbryanWyndon** [comment]
so theres no way to compile the game without downloading all the dependencies? Any workaround?

**Comment by ElbryanWyndon** [comment]
... i figured it out WHY DIDNT you TELL ME I DIDNT HAVE TO DOWNLOAD ALL OF IT TO COMPILE IT INN HTML-5

**Comment by ElbryanWyndon** [comment]
BUT now i have a new issue

**Comment by ElbryanWyndon** [comment]
im gona make a different issue for it
